public/app/features/playlist/PlaylistSrv.ts

```diff
@@ -78,17 +82,50 @@ export class PlaylistSrv {
     // setup location tracking
     this.locationListenerUnsub = locationService.getHistory().listen(this.locationUpdated);
 
-    return getBackendSrv()
-      .get(`/api/playlists/${playlistUid}`)
-      .then((playlist: any) => {
-        return getBackendSrv()
-          .get(`/api/playlists/${playlistUid}/dashboards`)
-          .then((dashboards: any) => {
-            this.dashboards = dashboards;
-            this.interval = rangeUtil.intervalToMs(playlist.interval);
-            this.next();
-          });
-      });
+    const searcher = getGrafanaSearcher();
+    let urls: string[] = [];
+    const playlist = await getBackendSrv().get<Playlist>(`/api/playlists/${playlistUid}`);
+    if (!playlist.items?.length) {
+      // alert
+      return;
+    }
+    this.interval = rangeUtil.intervalToMs(playlist.interval);
+    for (const item of playlist.items) {
+      if (!item.value) {
+        continue; // invalid item
+      }
+      const query: SearchQuery = { kind: ['dashboard'] };
+      switch (item.type) {
+        case 'dashboard_by_uid':
+          query.uid = [item.value];
+          break;
+
+        case 'dashboard_by_id':
+          query.uid = await getBackendSrv().get<string[]>(`/api/dashboards/ids/${item.value}`);
```

Comment:
That was my first attempt, but the logistics of getting the order consistent was more complex -- since we also have to insert the tags results potentially in the middle.

My general sense is that the APIs for this are fast enough that it does not matter. So having simpler APIs is likely best.

Reply:
I changed things so the frontend will now do the migration once from `dashboard_by_id` to `dashboard_by_uid` -- it now saves thins with uid 🎉 

It means the load dashboard backend call will not work, but that is OK because it is never called!